openshift/machine-config-operator
# Enable Alibaba Cloud provider instances with expected ProviderID

**- What I did**
This code enables the Alibaba Cloud provider platform on OpenShift.  The Alibaba provider requires that the instances have a specific format in which the [documentation](https://github.com/kubernetes/cloud-provider-alibaba-cloud/blob/master/docs/getting-started.md) states should be in `"${REGION}.${INSTANCE_ID}"`.

This patch follows the pattern for other providers which sets an environment variable named `KUBELET_NODE_NAME` in `/etc/systemd/system/kubelet.service.d/`. The environment variable is set through creating a service that upon start will query the metadata service for the region and the instanceid.

This pull request requires #2772.

**- How to verify it**
I was able to verify that this code works by deploying the following work:
- https://github.com/openshift/installer/pull/5018 with go.mod dependencies from https://github.com/openshift/installer/pull/5216
- Creating a custom build of the cluster-cloud-controller-manager-operator from Alibaba: https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/119
- Creating a custom build of the CCM from https://github.com/openshift/cloud-provider-alibaba-cloud with a specific fix for https://github.com/openshift/cloud-provider-alibaba-cloud/issues/10

**- Description for the changelog**
Enabling support for Alibaba Cloud provider node names
